File: Hue/ObjectModel/HueRuleInformation.cs
using System.Net.Mail;

namespace Hue.ObjectModel
{
    public class HueRuleInformation
    {
        public string id;
        public string name;
        public string owner;
        public string created;
        public string lasttriggered;
        public int timestriggered;
        public string status;
        public bool recycle;
        public HueRuleAction[] actions;
    }

    public class HueBasicRuleInformation
    {
        public string name;
        public HueRuleAction[] actions;

        public HueBasicRuleInformation(HueRuleInformation ruleInfo)
        {
            name = ruleInfo.name;
            actions = ruleInfo.actions;
        }
    }
}

File: Android/app/src/main/java/com/mikefilion/hueapp/ObjectModel/HueRuleInformation.java
package com.mikefilion.hueapp.ObjectModel;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.List;

/**
 * Created by mfilion on 2016-05-03.
 */
public class HueRuleInformation {

    public HueRuleInformation(JSONObject value, String ruleId) {
        id = ruleId;
        try {
            name = value.getString("name");
            owner = value.getString("owner");
            created = value.getString("created");
            lasttriggered = value.getString("lasttriggered");
            status = value.getString("status");
            timestriggered = value.getInt("timestriggered");
            recycle = value.getBoolean("recycle");
            JSONArray array = value.getJSONArray("actions");
            for (int i = 0; i < array.length(); ++i) {
                actions.add(new HueRuleAction(array.getJSONObject(i)));
            }
        } catch (JSONException e) {
            e.printStackTrace();
        }
    }

    public final String id;
    public String name;
    public String owner;
    public String created;
    public String lasttriggered;
    public String status;
    public int timestriggered;
    public boolean recycle;
    public List<HueRuleAction> actions;
}
